### ISSUE TITLE
v4.9 regression: segfaults on Mac when installed via Homebrew

### ISSUE THREAD

**Issue by mechfish** [eradman/entr#74]
Homebrew applies the `-Os` compiler optimization when installing `entr` on the Mac, which causes v4.9 to segfault. 

I eventually figured out that release 4.9 accidentally removed the call to `getrlimit()` from the Mac OS case, causing the code to try to read `open_max` from an uninitialized `struct rlimit rl`. This seems to work okay with optimizations off, because the uninitialized value is `0` and the "guard against unrealistic replies" sets `open_max = 65536` every time. But with optimizations on, it's a disaster.

I was able to fix this by copying the call to `getrlimit(RLIMIT_NOFILE, &rl)` from the BSD case to the Mac case, after which `entr` works even with optimizations on. However, I'm not making a PR because I don't understand what the code is doing or the semantics of `getrlimit`; I specifically don't understand why the Mac case reads `rlim_cur` while the BSD case reads `rlim_max`:

```
  # On the Mac it tries to execute this:
  open_max = min(65536, (unsigned)rl.rlim_cur);
  # But in BSD:
  open_max = (unsigned)rl.rlim_max;
  # Why? 
```

**Comment by eradman** [comment]
@mechfish, thanks for reporting this mistake--it does appear that `getrlimit()` was removed by accident. 

In 4.9 the code was reorganized in an attempt to make it more obvious what logic is used on each platform. I ended up switching to `rim_cur` for MacOS because `rlim_max` seems to be set to a bogus value on arm64:

```c
#include <stdio.h>
#include <unistd.h>
#include <sys/resource.h>

int main() {
        struct rlimit rl;
        getrlimit(RLIMIT_NOFILE, &rl);

        printf("rlim_cur: %lu\n", (unsigned long)rl.rlim_cur);
        printf("rlim_max: %lu\n", (unsigned long)rl.rlim_max);
        printf("_SC_OPEN_MAX: %lu\n", (unsigned long)sysconf(_SC_OPEN_MAX));
}
```

This prints the following when run on my M1:

```
rlim_cur: 256
rlim_max: 9223372036854775807
_SC_OPEN_MAX: 256
```

**Comment by mechfish** [comment]
Yeah, that `rlim_max` number certainly looks ridiculous, so your choice makes sense.

So the PR to fix this is probably as simple as I thought – I can put one together tomorrow if that would make your life easier. If you'd like to cut a bugfix release I can also try to expedite pushing that fix into Homebrew; Otherwise I should probably head over there and propose a temporary fix to the Homebrew recipe to turn off the compiler optimizations.

**Comment by eradman** [comment]
I pushed the fix in commit 468d77d. I'd like to see if any other oddities are reported before cutting a release.

If you have time, please propose a temporary fix in the Homebrew recipe--my impression is that this is how most people install `entr`.

**Comment by eradman** [comment]
entr-5.0 released today with this fix included
